packages/block-library/src/navigation-link/edit.js

```diff
@@ -262,7 +270,14 @@ export default function NavigationLinkEdit( {
 	const itemLabelPlaceholder = __( 'Add label…' );
 	const ref = useRef();
 	const linkUIref = useRef();
-	const prevUrl = usePrevious( url );
+
+	useEffect( () => {
+		// If the link was just added and label is url-like, focus and select the label text.
+		if ( isNewLink && isLabelURLLike ) {
+			// Focus and select the label text.
+			selectLabelText();
+		}
+	}, [] );
```

Comment:
I expect you want the dependencies to rerun this like it was before.

```suggestion
	}, [ isLabelURLLike, isNewLink ] );
```

Reply:
I only want this to run on mount.